web/src/beta/lib/core/engines/Cesium/Sketch/SurfaceMeasurement.tsx

```diff
@@ -0,0 +1,79 @@
+import {
+  CallbackProperty,
+  Cartesian3,
+  ClassificationType,
+  PolylineDashMaterialProperty,
+  type Color,
+  type PositionProperty,
+} from "@cesium/engine";
+import { useMemo, useRef, type FC } from "react";
+
+import { useConstant } from "@reearth/beta/utils/util";
+
+import { useVisualizer } from "../../../Visualizer";
+
+import { Entity, type EntityProps } from "./Entity";
+import { MeasurementText } from "./MeasurementText";
+import ScreenSpaceElement from "./ScreenSpaceElement";
+
+const MeasurementLine: FC<{
+  a: Cartesian3;
+  b: Cartesian3;
+  color?: Color;
+}> = ({ a, b, color }) => {
+  const controlPointsRef = useRef([a, b]);
+  controlPointsRef.current = [a, b];
+  const positionsProperty = useConstant(
+    () =>
+      new CallbackProperty(() => controlPointsRef.current, false) as unknown as PositionProperty,
+  );
+
+  const options = useMemo(
+    (): EntityProps => ({
+      polyline: {
+        positions: positionsProperty,
+        width: 1.5,
+        material: new PolylineDashMaterialProperty({
+          color,
+          dashLength: 8,
+        }),
+        classificationType: ClassificationType.TERRAIN,
+        clampToGround: true,
+      },
+    }),
+    [color, positionsProperty],
+  );
+
+  const visualizer = useVisualizer();
+  visualizer.current?.engine.requestRender();
```

Comment:
We should use `useContext` from [here](https://github.com/reearth/reearth/blob/3bb0a22483a8c75a80791ca5c434ac9ec15fd093/web/src/beta/lib/core/engines/Cesium/Feature/context.ts).
We have to think about relationship for each modules, but we should use the context for now. `useVisualizer` is for user of NLS.